VisualRust/Racer/RacerSingleton.cs

```diff
@@ -67,10 +71,53 @@ private void ReinitializeRacerPaths()
             else
                 racerPathForExecution = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "Racer", BundledRacerExecutable);
             // Same for custom RUST_SRC_PATH
-            if(GetVisualRustProperty<bool>(env, nameof(RustOptionsPage.UseCustomSources)))
-                racerSourcesLocation = GetVisualRustProperty<string>(env, nameof(RustOptionsPage.CustomSourcesPath));
-            else
-                racerSourcesLocation = null;
+            var type = GetVisualRustProperty<RustOptionsPage.RustSource>(env, nameof(RustOptionsPage.SourceType));
+            switch (type) {
+                case RustOptionsPage.RustSource.EnvVariable:
+                    racerSourcesLocation = null;
+                    break;
+                case RustOptionsPage.RustSource.Custom:
+                    racerSourcesLocation = GetVisualRustProperty<string>(env, nameof(RustOptionsPage.CustomSourcesPath));
+                    break;
+                default:
+                    racerSourcesLocation = GetRustcSysroot();
+                    if (racerSourcesLocation != null && !Directory.Exists(racerSourcesLocation))
+                        racerSourcesLocation = null;
+                    break;
+            }
+
+        }
+
+        private static String GetRustcSysroot()
```

Comment:
@xilec Unortunately it's not that easy.
@Vbif Yeah, I can't find any help in that direction either. So, if `.GetAwaiter().GetResult()` doesn't work, then just leave the code as is (including the synchronous `GetRustcSysroot()` method) and add a `TODO` comment to eventually replace this with `Cargo.GetSysrootAsync()`.